docs/admin/authentication.md

```diff
@@ -60,19 +59,213 @@ more secure modes described above easier to use.
+This token is a JSON Web Token (JWT) with well known fields, such as a user's
+email, signed by the server. When used as a bearer token, the apiserver can
+verify ID token's signature and determine the end users identity.
+
+To enable the plugin, pass the following required flags:
+
+* `--oidc-issuer-url` URL of the provider which allows the apiserver to discover
+public signing keys. Only URLs which use the `https://` scheme are accepted.
+* `--oidc-client-id` A client id that all tokens must be issued for.
+
+Importantly, the apiserver is not an OAuth2 client, rather it can only be
+configured to trust a single client. This allows the use of public providers,
+such as Google, without trusting credentials issued to third parties. Admins who
+wish utilize multiple OAuth clients should explore providers which support the
+`azp` (authorized party) claim, a mechanism for allowing one client to issue
+tokens on behalf of another.
+
+The plugin also accepts the following optional flags:
+
+* `--oidc-ca-file` Used by the apiserver to establish and verify the secure connection to the issuer.
+* `--oidc-username-claim` JWT claim to use as the user name. By default `sub`,
+which is expected to be a unique identifier of the end user. Admins can choose
+other claims, such as `email`, depending on their provider.
+* `--oidc-groups-claim` JWT claim to use as the user's group. If the claim is present
+it must be an array of strings.
+
+## Webhook Token Authentication
+
+Webhook authentication is a hook for verifying bearer tokens.
+
+* `--authentication-token-webhook-config-file` a kubeconfig file describing how to access the remote webhook service.
+* `--authentication-token-webhook-cache-ttl` how long to cache authentication decisions. Defaults to two minutes.
+
+The configuration file uses the [kubeconfig](/docs/user-guide/kubeconfig-file/)
+file format. Within the file "users" refers to the API Server webhook and
+"clusters" refers to the remote service. An example would be:
+
+```yaml
+# clusters refers to the remote service.
+clusters:
+  - name: name-of-remote-authn-service
+    cluster:
+      certificate-authority: /path/to/ca.pem         # CA for verifying the remote service.
+      server: https://authn.example.com/authenticate # URL of remote service to query. Must use 'https'.
+
+# users refers to the API Server's webhook configuration.
+users:
+  - name: name-of-api-server
+    user:
+      client-certificate: /path/to/cert.pem # cert for the webhook plugin to use
+      client-key: /path/to/key.pem          # key matching the cert
+
+# kubeconfig files require a context. Provide one for the API Server.
+current-context: webhook
+contexts:
+- context:
+    cluster: name-of-remote-authn-service
+    user: name-of-api-sever
+  name: webhook
+```
+
+When a client attempts to authenticate with the apiserver using a bearer token,
+the authentication webhook queries the remote service with a review object
+containing the token.
```

Comment:
The client has to provide an `Authorization: Bearer $TOKEN` HTTP header.  Kubernetes will not challenge request that lack such a header.